hydra-node/src/Hydra/Chain/Blockfrost/Client.hs

```diff
@@ -450,8 +451,14 @@ queryScript scriptHashTxt = do
 -- fact this is a single address query always.
 queryUTxO :: NetworkId -> [Address ShelleyAddr] -> BlockfrostClientT IO UTxO
 queryUTxO networkId addresses = do
-  let address' = Blockfrost.Address . serialiseAddress $ List.head addresses
-  utxoWithAddresses <- Blockfrost.getAddressUtxos address'
+  let address = List.head addresses
+  let address' = Blockfrost.Address . serialiseAddress $ address
+  utxoWithAddresses <-
+    Blockfrost.getAddressUtxos address'
+      `catchError` \err ->
```

Comment:
I think it would be a bit better if we case match on the exact BlockfrostError constructor instead of matching on strings. We might want to map more Blockfrost related errors into Hydra in the future so this proposed change would be more convenient for us.